Packs/Cofense-Intelligence/Integrations/CofenseIntelligenceV2/CofenseIntelligenceV2.py

```diff
@@ -0,0 +1,390 @@
+
+
+def create_threat_md_row(threat: Dict, severity_level: int = None):
+    threat_row = {"Threat ID": threat.get("id", ""),
+                  "Threat Types": "\n".join(
+                      [m.get("description", "") for m in threat.get("malwareFamilySet", [])]),
+                  "Executive Summary": threat.get("executiveSummary", ""),
+                  "Campaign": threat.get("label", ""),
+                  "Last Published": epochToTimestamp(threat.get("lastPublished"))}
+    if severity_level:
+        threat_row["Verdict"] = VERDICT.get(severity_level)
+    return threat_row
+
+
+def threats_analysis(threats: List, indicator: str, threshold: str):
+    threshold_score = SEVERITY_SCORE.get(threshold)
+    if not threshold_score:
+        raise Exception(
+            f'Cofense error: Invalid threshold value: {threshold}. Valid values are: None, Minor, Moderate or Major')
+
+    md_data = []
+    indicator_found = False
+    dbot_score = adjusted_score = 0
+
+    for threat in threats:
+        severity_level = 0
+        for block in threat.get('blockSet'):
+            if block.get('impact'):
+                threat_score: int = SEVERITY_SCORE.get(block.get('impact'), 0)
+                adjusted_score = 3 if threshold_score <= threat_score else threat_score
+                if block.get('data') == indicator:
+                    dbot_score = severity_level = adjusted_score
+                    indicator_found = True
+                    break
+            severity_level = max(severity_level, adjusted_score)
+
+        md_data.append(create_threat_md_row(threat, severity_level))
+
+        if not indicator_found:
+            dbot_score = max(dbot_score, severity_level)
+
+    return md_data, dbot_score
+
+
+def connectivity_testing(client: Client) -> str:
+    """Tests API connectivity and authentication'
+
+    Returning 'ok' indicates that the integration works like it is supposed to.
+    Connection to the service is successful.
+    Raises exceptions if something goes wrong.
+
+    :type client: ``Client``
+    :param Client: client to use
+
+    :return: 'ok' if test passed, anything else will fail the test.
+    :rtype: ``str``
+    """
+
+    message: str = ''
+    try:
+        client.threat_search_call()
+        message = 'ok'
+    except DemistoException as e:
+        if 'Forbidden' in str(e) or 'Authorization' in str(e):
+            message = 'Authorization Error: make sure user name and password are correctly set'
+        else:
+            raise e
+    return message
+
+
+def search_url_command(client: Client, args: Dict[str, Any], params) -> CommandResults:
+    url = args.get('url')
+    if not url:
+        raise ValueError('url not specified')
+    result = client.threat_search_call(url=url)
+
+    threats = result.get('data', {}).get('threats', [])
+    md_data, dbot_score = threats_analysis(threats, indicator=url, threshold=params.get('url_threshold'))
+    result = client.threat_search_call(url=url)
```

Comment:
why are you repeating the call from 124?

Reply:
fixed